File: src/main/java/com/geopark/web/model/entity/Geolandscape.java
package com.geopark.web.model.entity;

import com.alibaba.excel.annotation.ExcelProperty;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serializable;

/**
 * <p>
 * 地质遗迹景观
 * </p>
 *
 * @author lichee
 * @since 2019-09-04
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("geolandscape")
public class Geolandscape implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    @ExcelProperty(value = "id",index = 0)
    private Integer id;

    /**
     * 地质公园ID
     */
    @ExcelProperty(value = "公园编号",index = 1)
    private Integer parkId;

    /**
     * 地质遗迹统一编号
     */
    @ExcelProperty(value = "统一编号",index = 2)
    private String unitenumber;

    /**
     * 原编号
     */
    @ExcelProperty(value = "原编号",index = 3)
    private String originalnumber;

    /**
     * 地质遗迹名称
     */
    @ExcelProperty(value = "地质遗迹名称",index = 4)
    private String gname;

    /**
     * 地质遗迹原名称
     */
    @ExcelProperty(value = "原名称",index = 5)
    private String originalname;

    /**
     * 地质遗迹类型
     */
    @ExcelProperty(value = "遗迹类型",index = 6)
    private String type;

    /**
     * 地质遗迹地理位置
     */
    @ExcelProperty(value = "地理位置",index = 7)
    private String position;

    /**
     * 地质遗迹交通状况
     */
    @ExcelProperty(value = "交通状况",index = 8)
    private String traffic;

    /**
     * 经度
     */
    @ExcelProperty(value = "经度",index = 9)
    private String lat;

    /**
     * 纬度
     */
    @ExcelProperty(value = "纬度",index = 10)
    private String lng;

    /**
     * 海拔高度
     */
    @ExcelProperty(value = "海拔高度",index = 11)
    private String altitude;

    /**
     * 地质遗迹地质背景
     */
    @ExcelProperty(value = "地质背景",index = 12)
    private String background;

    /**
     * 地质遗迹特征
     */
    @ExcelProperty(value = "遗迹特征",index = 13)
    private String feature;

    /**
     * 评价级别
     */
    @ExcelProperty(value = "评价级别",index = 14)
    private String evaluation;

    /**
     * 保护级别
     */
    @ExcelProperty(value = "保护级别",index = 15)
    private String protection;

    /**
     * 观赏价值
     */
    @ExcelProperty(value = "观赏价值",index = 16)
    private String value;

    /**
     * 遗迹成因类型
     */
    @ExcelProperty(value = "成因类型",index = 17)
    private String causes;

    /**
     * 备注
     */
    @ExcelProperty(value = "备注",index = 18)
    private String note;

    /**
     * 照片
     */
    @ExcelProperty(value = "照片",index = 19)
    private String img;

}

File: src/main/resources/static/js/pkmanage/geolandscape.js
//公园概况

layui.use(['config', 'lichee', 'jquery', 'layer', 'table', 'form', 'laydate','imageUpload'], function () {

    var config = layui.config;
    var lichee = layui.lichee;
    var $ = layui.jquery;
    var layer = layui.layer;
    var table = layui.table;
    var form = layui.form;
    var laydate = layui.laydate;
    var imageUpload = layui.imageUpload;

    var tableInfo = table.render({
        elem: '#table',
        toolbar: '#toolbar',
        defaultToolbar: [],
        //width:'auto',
        url: '/geolandscape/page',
        title: '地质遗迹景观',
        page: true,
        headers: {Authorization: config.getToken()},
        request: config.request,
        parseData: config.parseData,
        response: config.response,
        cols: [[
            //{type: 'checkbox', fixed: 'left'},
            // {field: 'id', title: 'ID'},
            {field: 'parkId', align: 'center', sort: true, title: '地质公园ID'},
            {field: 'unitenumber', align: 'center', sort: true, title: '统一编号'},
            {field: 'gName', align: 'center', sort: true, title: '遗迹名称'},
            {field: 'originalname', align: 'center', sort: true, title: '原名称'},
            {field: 'type', align: 'center', sort: true, title: '遗迹类型'},
            {field: 'position', align: 'center', sort: true, title: '地理位置'},
            //{field: 'traffic', align: 'center', sort: true, title: '交通状况'},
            //{field: 'lat', align: 'center', sort: true, title: '经度'},
            //{field: 'lng', align: 'center', sort: true, title: '纬度'},
            {field: 'altitude', align: 'center', sort: true, title: '海拔高度'},
            //{field: 'background', align: 'center', sort: true, title: '地质背景'},
            {field: 'feature', align: 'center', sort: true, title: '特征'},
            {field: 'evaluation', align: 'center', sort: true, title: '评价级别'},
            {field: 'protection', align: 'center', sort: true, title: '保护级别'},
            {field: 'value', align: 'center', sort: true, title: '观赏价值'},
            //{field: 'causes', align: 'center', sort: true, title: '成因类型'},
            //{field: 'note', align: 'center', sort: true, title: '备注'},
            {fixed: 'right',align : 'center', title: '操作', toolbar: '#bar', width: 180}
        ]]
    });

    table.on('toolbar(tableFilter)', function (obj) {
        var checkStatus = table.checkStatus(obj.config.id);
        switch (obj.event) {
            case 'add':
                showEditModel();
                break;
            case 'getCheckLength':
                var data = checkStatus.data;
                layer.msg('选中了：' + data.length + ' 个');
                break;
            case 'isAll':
                layer.msg(checkStatus.isAll ? '全选' : '未全选');
                break;
        };
    });

    table.on('tool(tableFilter)', function (obj) {
        var data = obj.data;
        if (obj.event === 'del') {
            layer.confirm('确定要删除吗？', function (index) {
                layer.load(2);
                lichee.delete('/geolandscape/' + obj.data.id, {}, function () {
                    layer.closeAll('loading');
                    layer.msg('删除成功', {icon: 1});
                    obj.del();
                });
            });
        } else if (obj.event === 'edit') {
            showEditModel(data);
        }
    });

    var showEditModel = function (data) {
        layer.open({
            title: data ? '详情/修改' : '添加',
            type: 1,
            area: '800px',
            offset: '20px',
            content: $('#form-model').html(),
            success: function () {
                $('#form')[0].reset();
                lichee.get('/park/list', {async: false}, function (data) {
                    $('#parkId').vm({parks: data.result});
                    form.render('select');
                });
                imageUpload.init("uploadBtn", "img", "geolandscape");
                if (data) {
                    form.val('formFilter', data);
                    imageUpload.initImageList(data.img);
                }
                $('#form .close').click(function () {
                    layer.closeAll('page');
                });
            }
        });
    };

    var form = layui.form;
    form.on('submit(formSubmit)', function (data) {
        layer.load(2);
        if (data.field.id) {
            lichee.put('/geolandscape/' + data.field.id, {data: data.field}, function (res) {
                callFunction(res);
            });
        } else {
            lichee.post('/geolandscape', {data: data.field}, function (res) {
                callFunction(res);
            });
        }
        return false;
    });

    form.verify({
        commonLength: [/^[\S]{1,100}$/, '字符长度最大100'],
        descLength: [/^[\S]{0,180}$/, '字符长度最大180']
    });

    var callFunction = function (res) {
        layer.closeAll();
        if (res.status == 200) {
            layer.msg("修改成功",{icon: 1});
            tableInfo.reload({where: lichee.getSearchForm()});
        } else if(res.status == 201)  {
            layer.msg("添加成功",{icon: 1});
            tableInfo.reload({where: lichee.getSearchForm()});
        }
    };

    $('#searchButton').click(function () {
        tableInfo.reload({where: lichee.getSearchForm()});
    });

    laydate.render({
        elem: '#startTime',
        type: 'datetime',
        value: '',
        trigger: 'click'
    });

    $(".laydate").each(function (i,e) {
        laydate.render({
            elem: this,
            type: 'datetime',
            value: '',
            trigger: 'click'
        });
    });

});